src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/repository/impl/ords/DisputeRepositoryImpl.java

```diff
@@ -132,27 +130,47 @@ public Dispute saveAndFlush(Dispute entity) {
 		throw new NotYetImplementedException();
 	}
 
+	@Override
+	public void assignDisputeToUser(Long disputeId, String userName) {
+		assertNoExceptions(() -> violationTicketApi.v1AssignViolationTicketPost(disputeId, userName));
+	}
+
 	@Override
 	public void unassignDisputes(Date olderThan) {
-		SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DATE_TIME_FORMAT);
 		String dateStr = simpleDateFormat.format(olderThan);
 
-		// Propagate any ApiException to caller
-		ResponseResult result = violationTicketApi.v1UnassignViolationTicketPost(dateStr);
+		assertNoExceptions(() -> violationTicketApi.v1UnassignViolationTicketPost(dateStr));
+	}
+
+	@Override
+	public void setStatus(Long disputeId, DisputeStatus disputeStatus, String rejectedReason) {
+		assertNoExceptions(() -> violationTicketApi.v1ViolationTicketStatusPost(disputeId, disputeStatus.toShortName(), rejectedReason));
+	}
+
+	@Override
+	public void flushAndClear() {
+		// no-op. Not needed for ORDS.
+	}
+
+	/**
+	 * A helper method that will throw an appropriate InternalServerErrorException based on the ResponseResult. Any RuntimeExceptions throw will propagate up to caller.
+	 */
+	private void assertNoExceptions(Supplier<ResponseResult> m) {

```

Comment:
Very nice! So we're assuming that all requests will return a single type of ResponseResult except the ones that return an object JSON like ViolationTicket JSON I think.

Reply:
Yes, so far that seems to be the case.  All the endpoints I worked with so far all return a json object
{ status: "0", exception: "" }, so I renamed the DeleteResult to be more generic so it could be reused by other endpoints.